jme3-desktop/src/main/java/com/jme3/system/NativeLibraries.java

```diff
@@ -0,0 +1,170 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.system;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Defines default native libraries used by jMonkeyEngine.
+ *
+ * @author Ali-RS
+ */
+public enum NativeLibraries {
+
+    // Note: LWJGL 3 handles its native library extracting & loading using
+    // its own SharedLibraryLoader.
+
+    // LWJGL 2
+    Lwjgl(new LibraryInfo("lwjgl", libPath ->
+            // lwjgl handle loading by itself.
+            System.setProperty("org.lwjgl.librarypath",
+                    Paths.get(libPath).getParent().toAbsolutePath().toString()))
+            .addNativeVariant(Platform.Windows32, "lwjgl.dll")
+            .addNativeVariant(Platform.Windows64, "lwjgl64.dll")
+            .addNativeVariant(Platform.Linux32, "liblwjgl.so")
+            .addNativeVariant(Platform.Linux64, "liblwjgl64.so")
+            .addNativeVariant(Platform.MacOSX32, "liblwjgl.dylib")
+            .addNativeVariant(Platform.MacOSX64, "liblwjgl.dylib")
+    ),
+
+    // OpenAL for LWJGL 2
+    // For OSX: Need to add lib prefix when extracting
+    Openal(new LibraryInfo("openal")
+            .addNativeVariant(Platform.Windows32, "OpenAL32.dll")
+            .addNativeVariant(Platform.Windows64, "OpenAL64.dll")
+            .addNativeVariant(Platform.Linux32,   "libopenal.so")
+            .addNativeVariant(Platform.Linux64,   "libopenal64.so")
+            .addNativeVariant(Platform.MacOSX32,  "openal.dylib", "libopenal.dylib")
+            .addNativeVariant(Platform.MacOSX64,  "openal.dylib", "libopenal.dylib")
+    ),
+
+    // BulletJme
+    BulletJme(new LibraryInfo("bulletjme")
+            .addNativeVariant(Platform.Windows32, "native/windows/x86/bulletjme.dll", "bulletjme-x86.dll")
+            .addNativeVariant(Platform.Windows64, "native/windows/x86_64/bulletjme.dll", "bulletjme-x86_64.dll")
+            .addNativeVariant(Platform.Windows_ARM64, "native/windows/arm64/bulletjme.dll", "bulletjme-arm64.dll")
+            .addNativeVariant(Platform.Linux32, "native/linux/x86/libbulletjme.so", "libbulletjme-x86.so")
+            .addNativeVariant(Platform.Linux64, "native/linux/x86_64/libbulletjme.so", "libbulletjme-x86_64.so")
+            .addNativeVariant(Platform.Linux_ARM32, "native/linux/arm32/libbulletjme.so", "libbulletjme-arm32.so")
+            .addNativeVariant(Platform.Linux_ARM64, "native/linux/arm64/libbulletjme.so", "libbulletjme-arm64.so")
+            .addNativeVariant(Platform.MacOSX32, "native/osx/x86/libbulletjme.dylib", "libbulletjme-x86.dylib")
+            .addNativeVariant(Platform.MacOSX64, "native/osx/x86_64/libbulletjme.dylib", "libbulletjme-x86_64.dylib")
+            .addNativeVariant(Platform.MacOSX_ARM64, "native/osx/arm64/libbulletjme.dylib", "libbulletjme-arm64.dylib")
+    ),
+
+    // JInput
+    // For OSX: Need to rename extension jnilib -> dylib when extracting
```

Comment:
What is the `.jnilib` extension ?

Reply:
Is jinput outputing their native binaries with the `jnilib` extensions ?

Reply:
Yes, for macOS it is named `libjinput-osx.jnilib`. I do not know what it is.
The comment is from the old code.

Reply:
I see, no problem, you can resolve this.